source/UOXJSMethods.cpp

```diff
@@ -3642,36 +3650,44 @@ JSBool CBase_UseResource( JSContext *cx, JSObject *obj, uintN argc, jsval *argv,
 
 	UI32 amount		= static_cast<UI32>( JSVAL_TO_INT( argv[0] ));
 	UI16 realId		= static_cast<UI16>( JSVAL_TO_INT( argv[1] ));
-	UI16 itemColour = 0;
-	UI32 moreVal	= 0;
+	SI32 itemColour = 0;
+	SI64 moreVal	= -1;
```

Comment:
```suggestion
	SERIAL moreVal	= INVALIDSERIAL;
```
Might work better, like above?